File: stubs/csharp/src/BeastMode.RmDdd.Examples/Examples/ECommerce/Events/OrderLineRemovedEvent.cs
using System.Collections.Generic;
using BeastMode.RmDdd.Domain;

namespace BeastMode.RmDdd.Examples.ECommerce.Events
{
    /// <summary>
    /// Domain event fired when an order line is removed from an order.
    /// </summary>
    public class OrderLineRemovedEvent : DomainEvent
    {
        public ProductId ProductId { get; }
        
        public OrderLineRemovedEvent(OrderId orderId, ProductId productId) : base(orderId)
        {
            ProductId = productId;
        }
        
        public OrderId OrderId => (OrderId)AggregateId;
        
        public override IDictionary<string, object> GetEventData()
        {
            return new Dictionary<string, object>
            {
                ["order_id"] = OrderId.Value.ToString(),
                ["product_id"] = ProductId.Value.ToString()
            };
        }
    }
}

File: stubs/java/src/main/java/com/beastmode/rmddd/examples/ecommerce/OrderLineRemovedEvent.java
package com.beastmode.rmddd.examples.ecommerce;

import com.beastmode.rmddd.domain.DomainEvent;

import java.util.Map;

/**
 * Domain event fired when an order line is removed from an order.
 */
public class OrderLineRemovedEvent extends DomainEvent {
    
    private final ProductId productId;
    
    public OrderLineRemovedEvent(OrderId orderId, ProductId productId) {
        super(orderId);
        this.productId = productId;
    }
    
    public OrderId getOrderId() {
        return (OrderId) getAggregateId();
    }
    
    public ProductId getProductId() {
        return productId;
    }
    
    @Override
    public Map<String, Object> getEventData() {
        return Map.of(
            "order_id", getOrderId().getValue().toString(),
            "product_id", productId.getValue().toString()
        );
    }
}